File: JavaDBImplementation/database_insert.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

class database_insert {

    private static final String DB_URL = "jdbc:sqlite:database.db"; // to be edited when correct driver found
    
    public void addCourse(int courseID, String courseName, String courseCode) {
        try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("INSERT INTO courses VALUES(" + courseID + ", " + courseName + ", " + courseCode + ");");
        }
        catch (Exception e) {
            System.err.println(e);
        }
    }

    public void addModule(int moduleID, String moduleName, String moduleCode, int moduleCourseID) {
        try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("INSERT INTO modules VALUES(" + moduleID + ", " + moduleName + ", " + moduleCode + ", " + moduleCourseID + ");");
        }
        catch (Exception e) {
            System.err.println(e);
        }
    }

    public void addAssignment(int assignmentID, String assignmentName, String assignmentDeadline, double assignmentWeight, double gradeReceived, int assignmentModuleID) {
        try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("INSERT INTO assignments VALUES(" + assignmentID + ", " + assignmentName + ", " + assignmentDeadline + ", " + assignmentWeight + ", " + gradeReceived + ", " + assignmentModuleID +");");
        }
        catch (Exception e) {
            System.err.println(e);
        }
    }

    public void addStudySession(int sessionID, String startTime, String endTime, double duration, String location, int productivity, int sessionAssignmentID) {
        try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("INSERT INTO study_sessions VALUES(" + sessionID + ", " + startTime + ", " + endTime + ", " + duration + ", " + location + ", " + productivity + ", " + sessionAssignmentID +");");
        }
        catch (Exception e) {
            System.err.println(e);
        }
    }
}

File: Database/database_insert.py
import sqlite3

def addCourse(courseName, courseCode):
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    query = "INSERT INTO courses(course_name, course_code) VALUES('" + courseName + "', '" + courseCode + "');"
    cursor.execute(query)
    # Saves the changes and closes the connection
    conn.commit()
    conn.close()
    
def addModule(moduleName, moduleCode, moduleCourseID):
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    query = ("INSERT INTO modules(module_name, module_code, module_course_id) VALUES(" + moduleName + ", " + moduleCode + ", " + moduleCourseID + ");")
    cursor.execute(query)
    # Saves the changes and closes the connection
    conn.commit()
    conn.close()
    
def addAssignment(assignmentName, assignmentDeadline, assignmentWeight, gradeReceived, assignmentModuleID):
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    query = ("INSERT INTO assignments(assignment_name, assignment_deadline, assignment_weight, grade_received, assignment_module_id) VALUES(" + assignmentName + ", " + assignmentDeadline + ", " + assignmentWeight + ", " + gradeReceived + ", " + assignmentModuleID + ");")
    cursor.execute(query)
    # Saves the changes and closes the connection
    conn.commit()
    conn.close()

def addStudySession(startTime, endTime, duration, location, productivity, sessionAssignmentID):
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    query = ("INSERT INTO study_sessions(start, end, duration, location, productivity, session_assignment_id) VALUES(" + startTime + ", " + endTime + ", " + duration + ", " + location + ", " + productivity + ", " + sessionAssignmentID + ");")
    cursor.execute(query)
    # Saves the changes and closes the connection
    conn.commit()
    conn.close()